service/history/execution/context.go

```diff
@@ -198,6 +205,37 @@ func NewContext(
 		stats: &persistence.ExecutionStats{
 			HistorySize: 0,
 		},
+
+		appendHistoryNodesFn: func(ctx context.Context, domainID string, workflowExecution types.WorkflowExecution, request *persistence.AppendHistoryNodesRequest) (*persistence.AppendHistoryNodesResponse, error) {
+			throttleRetry := backoff.NewThrottleRetry(
+				backoff.WithRetryPolicy(common.CreatePersistenceRetryPolicy()),
+				backoff.WithRetryableError(persistence.IsTransientError),
+			)
+			return appendHistoryV2EventsWithRetry(ctx, shard, throttleRetry, domainID, workflowExecution, request)
+		},
+		createWorkflowExecutionFn: func(ctx context.Context, request *persistence.CreateWorkflowExecutionRequest) (*persistence.CreateWorkflowExecutionResponse, error) {
+			isRetryable := func(err error) bool {
+				if _, ok := err.(*persistence.TimeoutError); ok {
+					// TODO: is timeout error retryable for create workflow?
+					// if we treat it as retryable, user may receive workflowAlreadyRunning error
+					// on the first start workflow execution request.
+					return false
+				}
+				return persistence.IsTransientError(err)
+			}
+
+			throttleRetry := backoff.NewThrottleRetry(
+				backoff.WithRetryPolicy(common.CreatePersistenceRetryPolicy()),
```

Comment:
you could just add override for the policy instead of making the surrounding code an inline function. this inline function is not covered now